app/components/quick_text_input.js

```diff
@@ -59,9 +59,7 @@ export default class QuickTextInput extends React.PureComponent {
     }
 
     componentDidUpdate(prevProps, prevState) {
-        if (this.props.value !== this.storedValue) {
-            this.updateInputFromProps();
-        }
+        this.updateInputFromProps();
```

Comment:
`this.props.value !== this.storedValue` is always false since they are always equal on `handleChangeText`.  That's why the `QuickTextInput` component is not changing on update of props.value.

Reply:
Im not sure why HH had ot the other way

Reply:
It was that way because calling `this.updateInputFromProps` breaks certain keyboard layouts (eg Chinese and Japanese) when you're part way through typing something. The check of `this.props.value !== this.storedValue` is because we only need to call `this.updateInputFromProps` if  the value props changes due to something other than the user typing (like when we clear the input or load a message draft when switching channels).

Reply:
I think the problem here is that when the textbox becomes no longer editable, it switches back to the last value set as a prop or using `setNativeProps`. To avoid breaking the Chinese/Japanese behaviour, we could change the condition of the if statement to `this.props.value !== this.storedValue || !this.props.editable`

Reply:
You're right.  I did it and it worked.  Thanks!